### ISSUE TITLE
Update the Immer middleware Typescript example

### ISSUE THREAD

**Issue by Elia-Darwish** [pmndrs/zustand#338]
When updating to V9, the example shown in the docs will throw the following type error...

```
Conversion of type 'void extends ValidRecipeReturnTypePossiblyPromise<Draft<T>> ? (base: Immutable<Draft<T>>) => Draft<T> : never' to type '(state: T) => T' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.ts(2352)
```

With V9, it is possible now to pass the state generic directly to produce.

Before V9
```ts
import { State, StateCreator } from 'zustand'
import produce, { Draft } from 'immer'

const immer = <T extends State>(
  config: StateCreator<T, (fn: (draft: Draft<T>) => void) => void>
): StateCreator<T> => (set, get, api) =>
  config((fn) => set(produce(fn) as (state: T) => T), get, api)
```

After V9
```diff
import { State, StateCreator } from 'zustand'
import produce, { Draft } from 'immer'

const immer = <T extends State>(
  config: StateCreator<T, (fn: (draft: Draft<T>) => void) => void>
): StateCreator<T> => (set, get, api) =>
- config((fn) => set(produce(fn) as (state: T) => T), get, api)
+ config((fn) => set(produce<T>(fn)), get, api)
```

**Comment by dai-shi** [comment]
Oh, that looks nice. Would you mind opening a PR?

**Comment by Elia-Darwish** [comment]
I'm on it!!

**Comment by Elia-Darwish** [comment]
Closed by #339
